Drop Python 3.8

aaronliu0130 opened the pull request:
Pylint released their update on dedeprecating the getopt and optparse modules.Should merge this after we release 2.0.1.

aaronliu0130 committed:
drop Python 3.8

aaronliu0130 committed:
of couse you'd forget

aaronliu0130 commented:
Does not compute. Pylint 3.3.5 is available for PyPy 3.11 on Linux and macOS but not Windows‽

cclauss commented:
> Does not compute. Pylint 3.3.5 is available for PyPy 3.11 on Linux and macOS but not Windows‽

Yes, unfortunately correct... https://github.com/pylint-dev/pylint/pull/10286#pullrequestreview-2694823100

aaronliu0130 commented:
Pylint successfully installs and runs with PyPy3.11 on my Windows VM, but not on GitHub's apparently.

cclauss commented:
* pylint-dev/pylint#10288

aaronliu0130 committed:
Remove annotations imports from future

Also, remove ruff target-version property as it detects the target version from requires-python by default.

aaronliu0130 commented:
We could ignore the pypy3.11 issue for a while until release where maybe we can decide to just yank the pypy3.11 tests. Either way, this should be the next PR merged before any changelog-worthy changes so we have updated changelog and version entries.

aaronliu0130 committed:
bump version

cclauss commented on pyproject.toml:
```diff
@@ -43,7 +42,7 @@ dependencies = [  ]
   "mypy",
   "parameterized",
-  "pylint>=2.11",
+  "pylint>=3.3.5",
```
```suggestion
  "pylint==3.3.5",
```

cclauss committed:
pylint==3.3.5

cclauss committed:
pip install --editable ".[dev]"

cclauss commented on CHANGELOG.rst:
```diff
@@ -2,7 +2,12 @@
+2.1.0 (TBA)
+===========
+
+* Python versions less than 3.8 are no longer supported.
```
```suggestion
* Python versions less than 3.9 are no longer supported.
```

cclauss commented on cpplint.py:
```diff
@@ -61,7 +61,7 @@
 _valid_extensions: set[str] = set()
 
-__VERSION__ = "2.0.1"
+__VERSION__ = "2.1.0"
```
```suggestion
__VERSION__ = "2.0.1"
```

aaronliu0130 replied:
We do want a version bump for the next release, no? The trunk version is past the 2.0.1 release and shouldn't display 2.0.1 as its version.

cclauss commented on pyproject.toml:
```diff
@@ -43,7 +42,7 @@ dependencies = [  ]
   "mypy",
   "parameterized",
-  "pylint>=2.11",
+  "pylint==3.3.5",
```
```suggestion
  "pylint>=3.3.4",
```

aaronliu0130 replied:
I am surprised that this works since the dedeprecation only happens in pylint 3.3.5 and the pypy3.11 logs indicate that it installs 3.3.4.

cclauss committed:
Apply suggestions from code review

cclauss commented:
What is release vNEXT?  Is it v2.0.2, v2.1.0, or v3.0.0?  Different pull requests making different guesses is suboptimal (git conflicts).  It is best to make the vNEXT version number decision at release time.  v2.0.1 is preserved at:
* https://github.com/cpplint/cpplint/releases/tag/2.0.1
* https://github.com/cpplint/cpplint/tree/2.0.1

aaronliu0130 commented:
Thanks for the review! I would appreciate it if I or someone else had the chance to review your changes before they are unilaterally merged, though. (Not that that means a PR should be stuck just because the author didn't reply to changes; 24 hours should be enough.)
